File: solutions/2486. Append Characters to String to Make Subsequence/2486.cpp
// Time complexity: O(n)
// Space complexity: O(1)
class Solution {
 public:
  int appendCharacters(string_view s, string_view t) {
    int i = 0;

    for (const char c : s)
      if (c == t[i])
        if (++i == t.length())
          return 0;

    return t.length() - i;
  }
};

File: solutions/2486. Append Characters to String to Make Subsequence/2486.py
# Time complexity: O(n)
# Space complexity: O(1)
class Solution:
  def appendCharacters(self, s: str, t: str) -> int:
    i = 0
    for c in s:
      if i < len(t) and c == t[i]:
        i += 1
        if i == len(t):
          return 0

    return len(t) - i